tcp/clientconn.go

```diff
@@ -241,7 +240,11 @@ func (cc *ClientConn) Session() *Session {
 
 // Close closes connection without wait of ends Run function.
 func (cc *ClientConn) Close() error {
-	return cc.session.Close()
+	err := cc.session.Close()
+	if coapNet.IsCancelOrCloseError(err) {
```

Comment:
Hmm, something is closing the underlying connection so the atomic variable guard fails sometimes, I guess?